src/Sentry/buildTransitive/Sentry.targets

```diff
@@ -242,5 +250,119 @@
     <Warning Condition="'$(_SentryCLIExitCode)' != '0'" Text="Sentry CLI could not upload proguard mapping." />
   </Target>
 
+  <UsingTask TaskName="SentryGetReleaseFromInformationalVersion" TaskFactory="RoslynCodeTaskFactory" AssemblyFile="$(MSBuildToolsPath)\Microsoft.Build.Tasks.Core.dll">
```

Comment:
Is there a way for us to reuse the logic we use during runtime?

Reply:
I did a bit of experimenting and both the android build properties and the ios/macCatalyst information properties end up in the `AssemblyInformationVersionAttribute` anyway, so we can simply remove the target specific logic for those. 

I've also removed the code that tried to [get the version from the main assembly identity](https://github.com/getsentry/sentry-dotnet/blob/61cca84b41e2cdfecf04ec7cc6b103a1f2282bab/src/Sentry/buildTransitive/Sentry.targets#L330-L335). I'm not actually sure if/how that differs from `Assembly.GetName().Version?.ToString()` anyway (I think that is the main assembly identity) 

Finally, at run time we try to read the release from an environment variable so I changed the build targets to mimic this behaviour. 

That leaves us with:
| At build   | At runtime |
| ------------- | ------------- |
| `$(SentryRelease)` build property  |   |
|   | `SentryOptions.Release`  |
|  `SENTRY_RELEASE` Environment variable | `SENTRY_RELEASE` Environment variable  |
| `AssemblyInformationVersionAttribute`  | `AssemblyInformationVersionAttribute`  |
| `Assembly Name Version`  | `Assembly Name Version`  |

So they now both behave the same unless you override the release via the `$(SentryRelease)` build property (at build time) or via the `SentryOptions` (at runtime).

Reply:
Oh that's really really nice!